packages/xpetra/src/CrsMatrix/Xpetra_TpetraBlockCrsMatrix_def.hpp

```diff
@@ -486,41 +486,66 @@ namespace Xpetra {
                     size_t &NumEntries) const
     { 
         XPETRA_MONITOR("TpetraBlockCrsMatrix::getLocalRowCopy"); 
-        mtx_->getLocalRowCopy(LocalRow, Indices, Values, NumEntries); 
+        typename Tpetra::CrsGraph<LocalOrdinal,GlobalOrdinal,Node>::nonconst_local_inds_host_view_type indices("indices",Indices.size());
+        typename Tpetra::CrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>::nonconst_values_host_view_type values("values", Values.size());
+
+        mtx_->getLocalRowCopy(LocalRow, indices, values, NumEntries);
+        for (size_t i=0;i<NumEntries; ++i) {
+          Indices[i]=indices(i);
+          Values[i]=values(i);
+        }
+    }
+    template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+    void
+    TpetraBlockCrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>::
+    getLocalRowView(LocalOrdinal LocalRow, ArrayView< const LocalOrdinal > &Indices,
+                    ArrayView< const Scalar > &Values) const
+    {
+      XPETRA_MONITOR("TpetraBlockCrsMatrix::getLocalRowView");
+      typename Tpetra::CrsGraph<LocalOrdinal,GlobalOrdinal,Node>::local_inds_host_view_type indices;
+      typename Tpetra::CrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>::values_host_view_type values;
+
+      mtx_->getLocalRowView(LocalRow, indices, values);
+      Indices = ArrayView<const LocalOrdinal> (indices.data(), indices.extent(0));
+      Values = ArrayView<const Scalar> (values.data(), values.extent(0));
     }
 
 
     template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
     void 
     TpetraBlockCrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>::
     getGlobalRowView(GlobalOrdinal GlobalRow, 
-                     ArrayView< const GlobalOrdinal > &indices, 
-                     ArrayView< const Scalar > &values) const
+                     ArrayView< const GlobalOrdinal > &Indices,
+                     ArrayView< const Scalar > &Values) const
     { 
         XPETRA_MONITOR("TpetraBlockCrsMatrix::getGlobalRowView"); 
+        typename Tpetra::CrsGraph<LocalOrdinal,GlobalOrdinal,Node>::global_inds_host_view_type indices;
+        typename Tpetra::CrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>::values_host_view_type values;
+
         mtx_->getGlobalRowView(GlobalRow, indices, values);
+        Indices = ArrayView<const GlobalOrdinal> (indices.data(), indices.extent(0));
+        Values = ArrayView<const Scalar> (values.data(), values.extent(0));
     }
 
 
     template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
     void 
     TpetraBlockCrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>::
     getGlobalRowCopy(GlobalOrdinal GlobalRow, 
-                     const ArrayView< GlobalOrdinal > &indices, 
-                     const ArrayView< Scalar > &values, 
-                     size_t &numEntries) const
```

Comment:
Just to make sure regarding code style in Trilinos (or maybe only Xpetra): is first letter capitalization and mixed/camel case for subsequent words a requirement? Or just a matter of preference here?

Reply:
It's a mixed bag, i try to keep the same local scope,

MyClass and myVar seems to be the standard

Reply:
Okay noted, so `MyVar` for member vars, and `myVar` for local vars?